var/spack/repos/builtin/packages/faiss/package.py

```diff
@@ -20,29 +20,42 @@ class Faiss(AutotoolsPackage, CudaPackage):
     """
 
     homepage = "https://github.com/facebookresearch/faiss"
-    url = "https://github.com/facebookresearch/faiss/archive/v1.6.3.tar.gz"
+    url = "https://github.com/facebookresearch/faiss/archive/v1.7.2.tar.gz"
 
     maintainers = ["bhatiaharsh"]
 
-    version("1.6.3", sha256="e1a41c159f0b896975fbb133e0240a233af5c9286c09a28fde6aefff5336e542")
-    version("1.5.3", sha256="b24d347b0285d01c2ed663ccc7596cd0ea95071f3dd5ebb573ccfc28f15f043b")
+    # 1.7.2 was moved to cmake
+    version("1.7.2", sha256="d49b4afd6a7a5b64f260a236ee9b2efb760edb08c33d5ea5610c2f078a5995ec")
+
+    # previous versions used autotools
+    version(
+        "1.6.3",
+        sha256="e1a41c159f0b896975fbb133e0240a233af5c9286c09a28fde6aefff5336e542",
+        deprecated=True,
+    )
+    version(
+        "1.5.3",
+        sha256="b24d347b0285d01c2ed663ccc7596cd0ea95071f3dd5ebb573ccfc28f15f043b",
+        deprecated=True,
+    )
 
     variant("python", default=False, description="Build Python bindings")
-    variant("tests", default=False, description="Build Tests")
+    variant("shlib", default=False, description="Build shared library")
+    variant("tests", default=False, description="Build tests")
 
     conflicts("+tests", when="~python", msg="+tests must be accompanied by +python")
 
-    depends_on("python@3.7:", when="+python", type=("build", "run"))
+    depends_on("cmake@3.17:", type="build", when="@1.7.2:")
```

Comment:
Note that without #30738 also v1.6.3 and v1.5.2 will depend on `cmake`, since that is inherited from `CMakePackage`.

Reply:
okay, i was confused about this too. looks like i misinterpreted what Greg suggested